src/System.Private.ServiceModel/src/System/ServiceModel/Channels/HttpsChannelFactory.cs

```diff
@@ -242,19 +239,17 @@ private static void SetCertificate(HttpClientHandler handler, SecurityTokenConta
 
         private static void ValidateClientCertificate(X509Certificate2 certificate)
         {
-#if FEATURE_NETNATIVE
-            var query = new CertificateQuery
+            if (Fx.IsUap)
```

Comment:
Looks like you changed this entirely, not just doing the runtime check, I'll stop by and you can explain what the change is all about.